src/xdp/xsk.c

```diff
@@ -2947,52 +3082,50 @@ XskSockoptSetUmem(
         goto Exit;
     }
 
-    //
-    // If support is needed for kernel mode AF_XDP sockets, UMEM MDL setup
-    // needs more thought.
-    //
-    ASSERT(RequestorMode == UserMode);
-
-    Umem->Mapping.Mdl =
-        IoAllocateMdl(
-            Umem->Reg.Address,
-            (ULONG)Umem->Reg.TotalSize,
-            FALSE, // SecondaryBuffer
-            FALSE, // ChargeQuota
-            NULL); // Irp
-    if (Umem->Mapping.Mdl == NULL) {
-        Status = STATUS_INSUFFICIENT_RESOURCES;
-        goto Exit;
-    }
+    if (RequestorMode != KernelMode) {
+        Umem->Mapping.Mdl =
```

Comment:
Don't we always need an MDL, or at least need it if the bounce buffer is disabled and the underlying interface uses the MDL extension? Should be easy to repro a bugcheck if the `XskDisableTxBounce` regkey is set.

Reply:
Fixed. Needs a test case though